kernel/tests/dv.rs

```diff
@@ -47,3 +62,273 @@ fn non_dv_table() -> Result<(), Box<dyn std::error::Error>> {
     assert_eq!(total_rows, 10);
     Ok(())
 }
+
+/// Helper to extract scan files from a snapshot
+fn get_scan_files(
+    snapshot: Arc<Snapshot>,
+    engine: &dyn delta_kernel::Engine,
+) -> DeltaResult<Vec<FilteredEngineData>> {
+    let scan = snapshot.scan_builder().build()?;
+    let all_scan_metadata: Vec<_> = scan.scan_metadata(engine)?.collect::<Result<Vec<_>, _>>()?;
+
+    Ok(all_scan_metadata
+        .into_iter()
+        .map(|sm| sm.scan_files)
+        .collect())
+}
+
+/// End-to-end test that:
+/// 1. Creates a table with deletion vector support
+/// 2. Writes a parquet file with actual data rows
+/// 3. Creates deletion vectors marking specific rows as deleted
+/// 4. Writes the deletion vectors to a file using StreamingDeletionVectorWriter
+/// 5. Commits the deletion vectors in a transaction
+/// 6. Verifies that scanning only returns non-deleted rows
+#[tokio::test]
+async fn test_write_deletion_vectors_end_to_end() -> Result<(), Box<dyn std::error::Error>> {
+    let _ = tracing_subscriber::fmt::try_init();
+
+    // Create a table schema with id and value columns
+    let schema = Arc::new(StructType::try_new(vec![
+        StructField::nullable("id", DataType::INTEGER),
+        StructField::nullable("value", DataType::STRING),
+    ])?);
+
+    // Setup table with deletion vector support
+    let temp_dir = tempdir()?;
+    let base_url = url::Url::from_directory_path(temp_dir.path()).unwrap();
+    let (store, engine, table_url) = engine_store_setup("test_table", Some(&base_url));
+    let engine = Arc::new(engine);
+
+    // Create table with DV support (protocol 3/7 with deletionVectors feature)
+    create_table(
+        store.clone(),
+        table_url.clone(),
+        schema.clone(),
+        &[],
+        true, // use_37_protocol
+        vec!["deletionVectors"],
+        vec!["deletionVectors"],
+    )
+    .await?;
+
+    // Step 1: Create and write a parquet file with 10 rows
+    let data_batch = generate_batch(vec![
```

Comment:
exract the next few lines to a helper function that returns a random file name and takes data as input.  Then use the helper to write out two files instead of one.